### PR TITLE
Adapt to tokens with 2 decimals

### LINKED ISSUE #44: Validate new input/output token decimals
Some tokens, such as [GUSD](https://etherscan.io/token/0x056Fd409E1d7A124BD7017459dFEa2F387b6d5Cd) have just 2 decimal points.

Please assert in `updateInputToken` and `updateOutputToken` that it's not possible to add tokens with decimals less than 2. This should be covered by tests.

Also adapt the tests to use 2 decimals (GUSD)  as the minimum instead of the current 6 decimals (USDC) we use. I want to make sure swaps, exchange rates and add/remove liquidity functions all work well when we work with 2-decimal tokens.

### PR DESCRIPTION
**Pull request by esaulkov**
resolves #44

### CODE REVIEW

File: contracts/stablecoinswap.vy [hunk 1]
```diff
@@ -103,9 +103,9 @@ def removeLiquidity(token_address: address, stableswap_token_amount: uint256, er
 
     token_price: uint256 = self.tokenPrice(token_address)
     pool_size: uint256 = PriceOracle(self.priceOracleAddress).poolSize(self)
-    # erc20_token_amount = stableswapt_token_amount * pool_size / totalSupply / token_price * TOKEN_PRICE_MULTIPLIER
+    # erc20_token_amount = stableswap_token_amount * pool_size / totalSupply / token_price * TOKEN_PRICE_MULTIPLIER
     # It's better to divide at the very end for a higher precision
-    erc20_token_amount: uint256 = stableswap_token_amount * pool_size * TOKEN_PRICE_MULTIPLIER / token_price / self.totalSupply
+    erc20_token_amount: uint256 = stableswap_token_amount * pool_size / self.totalSupply
```
Comment (esaulkov) [inline contracts/stablecoinswap.vy]: I've moved conversion from stableswap tokens to output tokens below for a higher precision. During testing I became `erc20_token_amount` as `2003.996`. It was truncated to `2003`. After swap fee was substracted I got `2003 * 0.997 = 1996`. It is lower than expected amount: `2003.996 * 0.997 = 1997`.

File: tests/test_liquidity_pool.py [hunk 2]
```diff
@@ -219,7 +231,127 @@ def test_liquidity_pool(w3, contract, DAI_token, USDC_token, price_oracle, asser
     # Can add liquidity again after all liquidity is divested
     contract.addLiquidity(DAI_token.address, DAI_ADDED, DEADLINE, transact={'from': user1})
 
+def test_liquidity_gusd(w3, contract, DAI_token, GUSD_token, price_oracle, assert_fail):
```
Comment (blimp666) [inline tests/test_liquidity_pool.py]: It seems ```test_liquidity_gusd``` is quite similar to ```test_liquidity_pool``` (but uses GUSD instead of USDT). What is an idea of having both them? Is there anything that should be tested especially for USDT (6 decimals)? Or maybe it is possible to adapt tests we already had to use GUSD instead of USDT (and maybe to add few additional asserts if necessary) to keep a full cover with a lower number of lines?

File: tests/test_liquidity_pool.py [hunk 3]
```diff
@@ -60,29 +65,36 @@ def test_add_liquidity(w3, contract, DAI_token, USDC_token, price_oracle, assert
...
     price_oracle.updatePrice(DAI_token.address, int(NEW_DAI_PRICE * 10**8), transact={'from': owner})
-    assert contract.totalSupply() == DAI_ADDED * DAI_PRICE + USDC_ADDED * USDC_PRICE * 10**(18-6)
-    assert price_oracle.poolSize(contract.address) == DAI_ADDED * NEW_DAI_PRICE + int(USDC_ADDED * USDC_PRICE * 10**(18-6))
+    assert contract.totalSupply() == DAI_ADDED * DAI_PRICE + USDC_ADDED * USDC_PRICE * 10**(18-6) + GUSD_ADDED * GUSD_PRICE * 10**(18-2)
+    assert price_oracle.poolSize(contract.address) == DAI_ADDED * NEW_DAI_PRICE + int(USDC_ADDED * USDC_PRICE * 10**(18-6)) + int(GUSD_ADDED * GUSD_PRICE * 10**(18-2))
 
-    # following asserts are necessary to test a rounding
-    # add 1 DAI at the lowest level of precision (10^(-18) DAI token)
+    # # following asserts are necessary to test a rounding
+    # # add 1 DAI at the lowest level of precision (10^(-18) DAI token)
```
Comment (blimp666) [inline tests/test_liquidity_pool.py]: Is the second # symbol (here and below) intentional or just a typo?
